### ISSUE TITLE
Gnome desktop takes a long time to open after GDM login

### ISSUE THREAD

**Issue by cotswold66** [NixOS/nixpkgs#272909]
### Describe the bug
As the title says. Looking at the system log, it shows a lot of core dumps related to udisksd, specifically libnvme.

### Steps To Reproduce
1. Login to gnome from GDM
2. ...
3. ...

### Expected behavior
Login should be quick. No core dumps in system log.

### Additional context
I have researched the issue. This used to occur on both tumbleweed and archlinux using the same Dell XPS 13 9300 laptop. Both other distros have issued patches to correct. This [link](https://github.com/storaged-project/udisks/issues/1193) describes the issue and the two patches needed. I have appied both in an overlay and the issue goes away.

```console
  nixpkgs.overlays = [
    ( final: prev:
      {
        libnvme = prev.libnvme.overrideAttrs (old: {
          patches = (old.patches or []) ++ [
            (prev.fetchpatch {
              url = "https://github.com/linux-nvme/libnvme/pull/727.patch";
              hash = "sha256:0SrR1++QTXNq+X4YOpW3JzEjFdSSBdqPl/xI2zdcvTs=";
            })
          ];
        });
      } )
    ( final: prev:
      {
        libblockdev = prev.libblockdev.overrideAttrs (old: {
          patches = (old.patches or []) ++ [
            (prev.fetchpatch {
              url = "https://github.com/storaged-project/libblockdev/pull/969.patch";
              hash = "sha256:LHeotKzcRDdT/GhH3JdVjX/7ZMN1ghllYuaxPYsCZMY=";
            })
          ];
        });
      } )
  ];

```

### Notify maintainers

@fogti @vifino @johnazoidberg

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.65, NixOS, 23.11 (Tapir), 23.11.20231207.b4372c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by cotswold66** [comment]
Hi

Although I applied the patch to libblockdev 3.0.3, I believe it should also
be applied to 3.0.4. At least, they did in the archlinux repos.

Thanks.
--
John Lord
Tel: +1 512 550 4917

**Comment by vifino** [comment]
libblockdev v3.0.4 contains the PR that fixes the issue in libblockdev - see the changelog: https://github.com/storaged-project/libblockdev/commit/ec0120de06f28ecfc10953b712f7f42735787bcf

**Comment by cotswold66** [comment]
Great. I really appreciate you working on this fix.

**Comment by vifino** [comment]
Hey. It's been a while, but the PR's have been merged, it was fixed on unstable for a while, but will be fixed in a few days on 23.11 when nixos-23.11 advances.

**Comment by cotswold66** [comment]
Awesome, thanks for the update! This will save me some compile time.
